cpp/include/milvus-storage/reader_c.h

```diff
@@ -0,0 +1,206 @@
+  char* value;  ///< Property value (caller owns memory)
+} ReadProperty;
+
+/// C struct for read properties (array of key-value pairs)
+typedef struct {
+  ReadProperty* properties;  ///< Array of property key-value pairs (caller owns memory)
+  size_t count;              ///< Number of properties in the array
+} ReadProperties;
+
+/**
+ * @brief Creates default read properties (empty)
+ *
+ * @param properties Output parameter for default properties (caller must free)
+ */
+void read_properties_default(ReadProperties* properties);
+
+/**
+ * @brief Creates read properties from key-value arrays
+ *
+ * @param keys Array of property keys
+ * @param values Array of property values
+ * @param count Number of key-value pairs
+ * @param properties Output parameter for created properties (caller must free)
+ */
+void read_properties_create(const char* const* keys,
+                            const char* const* values,
+                            size_t count,
+                            ReadProperties* properties);
+
+/**
+ * @brief Gets a property value by key
+ *
+ * @param properties Properties to search
+ * @param key Property key to find
+ * @return Property value if found, NULL otherwise (do not free)
+ */
+const char* read_properties_get(const ReadProperties* properties, const char* key);
+
+/**
+ * @brief Frees memory allocated for ReadProperties
+ *
+ * @param properties Properties to free
+ */
+void read_properties_free(ReadProperties* properties);
+
+// ==================== ChunkReader C Interface ====================
+
+/// Opaque handle for ChunkReader
+typedef void* ChunkReaderHandle;
+
+/**
+ * @brief Maps row indices to their corresponding chunk indices
+ *
+ * @param reader ChunkReader handle
+ * @param row_indices Array of global row indices to map
+ * @param num_indices Number of indices in the array
+ * @param chunk_indices Output array of chunk indices (caller must free)
+ * @param num_chunk_indices Output number of chunk indices
+ */
+void get_chunk_indices(ChunkReaderHandle reader,
+                       const int64_t* row_indices,
+                       size_t num_indices,
+                       int64_t** chunk_indices,
+                       size_t* num_chunk_indices);
+
+/**
+ * @brief Retrieves a single chunk by its index
+ *
+ * @param reader ChunkReader handle
+ * @param chunk_index Zero-based index of the chunk to retrieve
+ */
+void get_chunk(ChunkReaderHandle reader, int64_t chunk_index);
+
+/**
+ * @brief Retrieves multiple chunks by their indices with optional parallel processing
+ *
+ * @param reader ChunkReader handle
+ * @param chunk_indices Array of chunk indices to retrieve
+ * @param num_indices Number of indices in the array
+ * @param parallelism Number of threads to use for parallel reading
+ * @param arrays Output array of RecordBatch handles (caller must free)
+ * @param num_arrays Output number of record batches
+ */
+void get_chunks(ChunkReaderHandle reader,
```

Comment:
if we got a exception in the C++ layer, `try...catch` in the extern "C" function. How does the caller know what specific error occurred?

In C, used to return a `int`, 0 eq success, neg value is the error code. we can provider a C function `char *get_errmsg(int errcode);` to tell caller the details of error.